osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinnerApproachCircle.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Skinning.Legacy
+{
+    public class LegacySpinnerApproachCircle : CompositeDrawable
+    {
+        private DrawableSpinner drawableSpinner;
+
+        [CanBeNull]
+        private Sprite approachCircle;
+
+        [BackgroundDependencyLoader]
+        private void load(DrawableHitObject drawableHitObject, ISkinSource source)
+        {
+            drawableSpinner = (DrawableSpinner)drawableHitObject;
+
+            AutoSizeAxes = Axes.Both;
+
+            var spinnerProvider = source.FindProvider(s =>
+                s.GetTexture("spinner-circle") != null ||
+                s.GetTexture("spinner-top") != null);
```

Comment:
Move to `LegacyNew` and `LegacyOld` spinner implementation. The old will *always* provide an approach circle. The new will only check against the single relevant texture `spinner-top`.

Probably make `ApproachCircle` itself non-null as well, putting an empty drawable in its place to make interface consumption easier.